crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1198,6 +1208,14 @@ impl<'src> Parser<'src> {
         if has_bytes {
             match byte_literal_count.cmp(&strings.len()) {
                 Ordering::Less => {
+                    // TODO(dhruvmanila): This is not an ideal recovery because the parser
+                    // replaces the byte literals with an invalid string literal node. Any
+                    // downstream tools can extract the raw bytes from the range.
+
+                    // test_err mixed_bytes_and_non_bytes_literals
+                    // 'first' b'second'
+                    // f'first' b'second'
+                    // 'first' f'second' b'third'
```

Comment:
Yeah. We probably want to convert the node to a string and mark it as "errornous". We might also need to be slightly more clever by either marking the string or byte literals as errors, depending on which nodes are fewer. But that's something we can do later.

Reply:
Yeah, I had a similar idea. I'll add this to the TODO comment.